File: src/main/java/io/dddrive/ddd/model/enums/CodeAggregateTypeEnum.java

package io.dddrive.ddd.model.enums;

import org.springframework.stereotype.Component;

import io.dddrive.app.service.api.impl.Enumerations;
import io.dddrive.enums.model.base.EnumerationBase;

@Component("codeAggregateTypeEnum")
public class CodeAggregateTypeEnum extends EnumerationBase<CodeAggregateType> {

	private static CodeAggregateTypeEnum INSTANCE;

	protected CodeAggregateTypeEnum(Enumerations enums) {
		super(CodeAggregateType.class, enums);
		INSTANCE = this;
	}

	public static CodeAggregateTypeEnum getInstance() {
		return INSTANCE;
	}

	public void addItem(CodeAggregateType item) {
		super.addItem(item);
	}

	public static CodeAggregateType getAggregateType(String aggregateTypeId) {
		return INSTANCE.getItem(aggregateTypeId);
	}

}

File: fm-dddrive/src/main/java/io/dddrive/ddd/model/enums/CodeAggregateTypeEnum.kt
package io.dddrive.ddd.model.enums

import io.dddrive.enums.model.base.EnumerationBase

class CodeAggregateTypeEnum : EnumerationBase<CodeAggregateType>(CodeAggregateType::class.java) {

	init {
		instance = this
	}

	companion object {

		lateinit var instance: CodeAggregateTypeEnum

		@JvmStatic
		fun getAggregateType(aggregateTypeId: String): CodeAggregateType = instance.getItem(aggregateTypeId)

	}

}
